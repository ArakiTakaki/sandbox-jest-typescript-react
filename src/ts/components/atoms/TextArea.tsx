import React, { ChangeEventHandler } from 'react';

interface Props {
  value: string;
  eventChange?: ChangeEventHandler;
  children?: string;
}
const TextArea = (props: Props) => {
  const { value, eventChange, children } = props;
  return (
    <div>
      InputType:{children}
      <input type='text' value={value} onChange={eventChange} />
    </div>
  );
};

export default TextArea;
