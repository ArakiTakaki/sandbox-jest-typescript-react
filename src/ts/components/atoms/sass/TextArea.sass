.textarea
  font-size: 20px
  width: 2000px