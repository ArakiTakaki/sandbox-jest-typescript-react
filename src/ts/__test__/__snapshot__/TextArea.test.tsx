import React from "react";
import TextArea from "../../components/atoms/TextArea";
import renderer from "react-test-renderer";

test("renders correctly", () => {
  const component = renderer.create(
    <TextArea value="sample value">test</TextArea>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
