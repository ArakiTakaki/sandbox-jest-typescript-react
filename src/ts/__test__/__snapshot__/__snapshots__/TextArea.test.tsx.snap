// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  test
  <input
    type="text"
    value="sample value"
  />
  ;
</div>
`;
